VideoLocker/src/main/java/org/edx/mobile/view/CourseDashboardFragment.java

```diff
@@ -30,111 +28,124 @@
-                if ( courseData != null )
-                    environment.getRouter().showCourseAnnouncement(getActivity(), courseData);
-            }
-        });
+        if (IS_COURSEWARE_ACCESSIBLE) {
+            //Implementation Note: - we can create a list view and populate the list.
+            //but as number of rows are fixed and each row is different. the only common
+            //thing is UI layout. so we reuse the same UI layout programmatically here.
+            LayoutInflater inflater = LayoutInflater.from(getActivity());
+            ViewHolder holder = createViewHolder(inflater, parent);
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_list_alt);
+            holder.titleView.setText(R.string.courseware_title);
+            holder.subtitleView.setText(R.string.courseware_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    environment.getRouter().showCourseContainerOutline(getActivity(), courseData);
+                }
+            });
+
+            holder = createViewHolder(inflater, parent);
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_comments_o);
+            holder.titleView.setText(R.string.discussion_title);
+            holder.subtitleView.setText(R.string.discussion_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    environment.getRouter().showCourseDiscussionTopics(getActivity(), courseData);
+                }
+            });
+
+            holder = createViewHolder(inflater, parent);
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_file_text_o);
+            holder.titleView.setText(R.string.handouts_title);
+            holder.subtitleView.setText(R.string.handouts_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (courseData != null)
+                        environment.getRouter().showHandouts(getActivity(), courseData);
+                }
+            });
+
+            holder = createViewHolder(inflater, parent);
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_bullhorn);
+            holder.titleView.setText(R.string.announcement_title);
+            holder.subtitleView.setText(R.string.announcement_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (courseData != null)
+                        environment.getRouter().showCourseAnnouncement(getActivity(), courseData);
+                }
+            });
+        } else {
+            errorText.setText(courseData.getCourse().getCoursewareAccess().getUser_message());
```

Comment:
We need to use a locally localized string here I guess, as Akiva mentioned in the comments on the issue.